src/services/bundle_repository.js

```diff
@@ -124,11 +124,32 @@ export default class BundleRepository {
   }
 
   async removeBundle(bundleId) {
-    await this.db.collection('bundle_metadata').deleteMany({bundleId});
     const cursor = await this.bundlesBucket.find({filename: bundleId});
     while (await cursor.hasNext()) {
       const {_id: id} = await cursor.next();
       await this.bundlesBucket.delete(id);
     }
+    await this.setBundleRepository(bundleId, BundleStatuses.expendable);
+  }
+
+  async findOutdatedBundles() {
+    await this.db.collection('bundle_metadata').updateMany({
+      $or: [
+        {repository: {$exists: false}},
+        {'repository.holdUntil': {$lt: new Date()}}
+      ]
+    }, {$set: {repository: {status: BundleStatuses.cleanup}}});
+  }
+
+  async cleanupBundles() {
+    const cursor = await this.db.collection('bundle_metadata').find({'repository.status': BundleStatuses.cleanup}, {projection: {bundleId: 1, _id: 0}});
```

Comment:
Maybe also find `expendable` ones? Just in case they weren't properly deleted.

Reply:
Amount of expendables will increase with each iteration. Also expendable status is set after bundle was removed, so realistically I don't see a case when this happens

Reply:
Makes sense